### PR TITLE
Update IteratorBenchmark.java

### PR DESCRIPTION
**Pull request by fvarg00**
Refer to objects by their interfaces(Item 52)

### CODE REVIEW

**Comment by fvarg00** [comment]
I signed it!

File: guava-tests/benchmark/com/google/common/collect/IteratorBenchmark.java [hunk 1]
```diff
@@ -32,7 +33,7 @@
 
   // use concrete classes to remove any possible polymorphic overhead?
   Object[] array;
-  ArrayList<Object> arrayList;
+  List<Object> arrayList;
   LinkedList<Object> linkedList;
```
Comment (jbduncan) [inline guava-tests/benchmark/com/google/common/collect/IteratorBenchmark.java]: How about this `LinkedList` as well?

Comment (perceptron8) [reply guava-tests/benchmark/com/google/common/collect/IteratorBenchmark.java]: How about this "use concrete classes to remove any possible polymorphic overhead?" comment? ;)

Comment (jbduncan) [reply guava-tests/benchmark/com/google/common/collect/IteratorBenchmark.java]: Ah, I missed that. Silly me. :P

Comment (fvarg00) [reply guava-tests/benchmark/com/google/common/collect/IteratorBenchmark.java]: I shall refactor to remote all concrete classes. Thanks @perceptron8

Comment (jbduncan) [reply guava-tests/benchmark/com/google/common/collect/IteratorBenchmark.java]: @fvarg00 Uh, I _think_ what the comment is saying is that concrete classes are purposely used _instead_ of the `List` interface to "remove any possible polymorphic overhead" (not entirely sure what that means), so AFAICT this PR would probably not be accepted even if the `LinkedList` is refactored...

Comment (fvarg00) [reply guava-tests/benchmark/com/google/common/collect/IteratorBenchmark.java]: @jbduncan - thanks for your comment. It makes sense now.

**Pull request closed without merging**